lisp/ess-help.el

```diff
@@ -56,6 +60,14 @@
 ;;;; * The major mode ess-help-mode
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(defun ess--set-help-major-mode (&optional dialect)
+  "Determine which help major mode to call, and call it.
+DIALECT defaults to `ess-dialect'."
+  (let ((ess-dialect (or dialect ess-dialect)))
+    (cond ((string= ess-dialect "R") (require 'ess-r-mode) (ess-r-help-mode))
```

Comment:
After this one is merged we should refactor such conds into cl-generics dispatching on ess-dialect. Coresponding functionality of cl-generic was introduced in e25.

Reply:
Yes. Depending on how attached you are, we could actually port from ess-generics and use cl-generics instead. I've played around a bit with porting the functions we have right now and it's not terribly difficult.

Reply:
Looks like there is some telepathy going on ;). Yes, no attachment.  cl-generic it is.